bundle/edu.gemini.shared.util/src/main/java/edu/gemini/shared/util/immutable/ImCollections.java

```diff
@@ -215,6 +217,11 @@ public Object max(final Comparator<? super Object> c) {
         }
 
         @Override
+        public <K> HashMap<K, ImList<Object>> groupBy(final Function1<? super Object, K> f) {
+            return new HashMap<>();
```

Comment:
That's what I initially tried but I'm getting:

```no instance(s) of type variable(s) K,V exist so that java.util.Map<K,V> conforms to java.util.HashMap<K,edu.gemini.shared.util.immutable.ImList<java.lang.Object>>```